test/python/primitives/test_backend_estimator_dev.py

```diff
@@ -0,0 +1,924 @@
+    IXII.measure(2, 0)
+    yield ["IXII", "IIII"], IXII
+
+
+def circuit_composition_examples() -> Iterator[
+    tuple[QuantumCircuit, QuantumCircuit, QuantumCircuit]
+]:
+    """Generator of base and measurement circuits, and respective composition.
+
+    Yields:
+        - Transpiled base circuit: with a `final_layout` entry in its metadata
+        - Measurement circuit: before transpilation (i.e. no layout applied)
+        - Transpiled measurement circuit: with `final_layout` applied
+    """
+    target_qubits = 2
+    layout_intlist = (0,)
+    measured_qubits = (0,)
+    yield build_composition_data(target_qubits, layout_intlist, measured_qubits)
+
+    target_qubits = 2
+    layout_intlist = (1,)
+    measured_qubits = (0,)
+    yield build_composition_data(target_qubits, layout_intlist, measured_qubits)
+
+    target_qubits = 2
+    layout_intlist = (1, 0)
+    measured_qubits = (1,)
+    yield build_composition_data(target_qubits, layout_intlist, measured_qubits)
+
+    target_qubits = 2
+    layout_intlist = (1, 0)
+    measured_qubits = (0, 1)
+    yield build_composition_data(target_qubits, layout_intlist, measured_qubits)
+
+    target_qubits = 3
+    layout_intlist = (2,)
+    measured_qubits = (0,)
+    yield build_composition_data(target_qubits, layout_intlist, measured_qubits)
+
+    target_qubits = 3
+    layout_intlist = (2, 0)
+    measured_qubits = (0, 1)
+    yield build_composition_data(target_qubits, layout_intlist, measured_qubits)
+
+    target_qubits = 3
+    layout_intlist = (1, 2)
+    measured_qubits = (1,)
+    yield build_composition_data(target_qubits, layout_intlist, measured_qubits)
+
+    target_qubits = 3
+    layout_intlist = (1, 0, 2)
+    measured_qubits = (1, 2)
+    yield build_composition_data(target_qubits, layout_intlist, measured_qubits)
+
+    target_qubits = 4
+    layout_intlist = (0, 1, 2)
+    measured_qubits = (0, 1)
+    yield build_composition_data(target_qubits, layout_intlist, measured_qubits)
+
+    target_qubits = 4
+    layout_intlist = (1, 0, 3)
+    measured_qubits = (1, 2)
+    yield build_composition_data(target_qubits, layout_intlist, measured_qubits)
+
+    target_qubits = 4
+    layout_intlist = (2, 1, 3, 0)
+    measured_qubits = (0, 1, 2)
+    yield build_composition_data(target_qubits, layout_intlist, measured_qubits)
+
+    target_qubits = 4
+    layout_intlist = (0, 2, 1, 3)
+    measured_qubits = (0, 1, 2, 3)
+    yield build_composition_data(target_qubits, layout_intlist, measured_qubits)
+
+
+def build_composition_data(
+    target_qubits: int, layout_intlist: Sequence[int], measured_qubits: Sequence[int]
+) -> tuple[QuantumCircuit, QuantumCircuit, QuantumCircuit]:
+    """Build base and measurement circuits, and respective composition.
+
+    Args:
+        - target_qubits: the number of qubits to target during transpilation
+        - layout_intlist: indices to map virtual qubits to during transpilaiton
+        - measured_qubits: virtual qubits to measure
```

Comment:
Very unimportant, but if you were aiming for normal docstring syntax, there's no `- ` before each argument in a section list like this.

Reply:
Oh! My bad! You are 100% correct